<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>
  <!-- Axes -->
  <node pkg="vrep_vsv_driver" type="vsv_arm_ik.py" name="arm_ik" output="screen">
    <param name="vsv_name" value="VSV"/>
    <param name="max_velocity" value="0.5"/>
    <remap from="/arm_ik/joint_command" to="/VSV/aggregated/command"/>
    <remap from="/arm_ik/joint_state" to="/VSV/aggregated/state"/>
  </node>

  <node pkg="vrep_vsv_driver" type="vsv_driver.py" name="vsv_driver" output="screen">
    <param name="vsv_name" value="VSV"/>
    <param name="min_radius" value="5.0"/>
  </node>

  <node pkg="vrep_vsv_driver" type="vsv_arm.py" name="vsv_arm" output="screen">
    <param name="vsv_name" value="VSV"/>
  </node>

  <node pkg="rostopic" type="rostopic" name="shore_setpoint" args="pub /lateral_distance/setpoint std_msgs/Float64 2.0"/>

  <node pkg="pid" type="controller" name="shore_controller" output="screen" >
    <param name="Kp" value="1.0" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.1" />
    <param name="upper_limit" value="10" />
    <param name="lower_limit" value="-10" />
    <param name="windup_limit" value="10" />
    <param name="max_loop_frequency" value="10.0" />
    <param name="min_loop_frequency" value="10.0" />
    <remap from="state" to="/lateral_distance/distance" />
    <remap from="setpoint" to="/lateral_distance/setpoint" />
    <remap from="control_effort" to="/lateral_distance/control_effort" />
  </node>

  <node pkg="vrep_vsv_driver" type="lateral_distance.py" name="lateral_distance" />

  <node pkg="vrep_vsv_driver" type="twist_robot.py" name="twist_robot" />

  <node pkg="rostopic" type="rostopic" name="arm_setpoint" args="pub /vertical_distance/setpoint std_msgs/Float64 0.4"/>

  <node pkg="pid" type="controller" name="arm_controller" output="screen" >
    <param name="Kp" value="0.5" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.1" />
    <param name="upper_limit" value="0.5" />
    <param name="lower_limit" value="-0.5" />
    <param name="windup_limit" value="0.5" />
    <param name="max_loop_frequency" value="10.0" />
    <param name="min_loop_frequency" value="10.0" />
    <remap from="state" to="/vertical_distance/distance" />
    <remap from="setpoint" to="/vertical_distance/setpoint" />
    <remap from="control_effort" to="/vertical_distance/control_effort" />
  </node>

  <!-- <node pkg="rqt_plot" type="rqt_plot" name="rqt_plot" args="/lateral_distance/setpoint /lateral_distance/distance/data" /> -->
  <!-- <node pkg="rqt_plot" type="rqt_plot" name="rqt_plot" args="/vertical_distance/setpoint /vertical_distance/distance/data" /> -->

  <node pkg="vrep_vsv_driver" type="vertical_distance.py" name="vertical_distance" />

  <node pkg="vrep_vsv_driver" type="mine_detector.py" name="mine_detector" />
  <!-- <node pkg="vrep_vsv_driver" type="twist_arm.py" name="twist_arm" /> -->

</launch>
